x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/EnrichStoreCrudTests.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.enrich;
+
+import org.elasticsearch.ResourceNotFoundException;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.xpack.core.enrich.EnrichPolicy;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.elasticsearch.xpack.enrich.EnrichPolicyTests.randomEnrichPolicy;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.nullValue;
+
+public class EnrichStoreCrudTests extends EnrichStoreTests {
```

Comment:
Can all these tests just stay in `EnrichStoreTests`?

Reply:
if they do and another class inherits from EnrichStoreTests, wont they be run in duplicate? Currently that transport test is a child of `EnrichStoreTests`

Reply:
ah, I missed that. Maybe then rename `EnrichStoreTests` to `AbstractEnrichTestCase`?

Reply:
this would make more sense

Reply:
4af9656